sql/core/src/test/scala/org/apache/spark/sql/internal/ExecutorSideSQLConfSuite.scala

```diff
@@ -125,6 +125,39 @@ class ExecutorSideSQLConfSuite extends SparkFunSuite with SQLTestUtils {
     val e = intercept[SparkException](dummyQueryExecution1.toRdd.collect())
     assert(e.getCause.isInstanceOf[NoSuchElementException])
   }
+
+  test("SPARK-30556 propagate local properties to subquery execution thread") {
+    withSQLConf("spark.sql.subquery.maxThreadThreshold" -> "1") {
+      spark.sparkContext.setLocalProperty("spark.sql.y", "e")
+      Seq(true)
+        .toDF()
+        .createOrReplaceTempView("l")
+      Seq(true)
+        .toDF()
+        .mapPartitions { _ =>
+          val conf = SQLConf.get
+          conf.isInstanceOf[ReadOnlySQLConf] && conf.getConfString("spark.sql.y") == "e" match {
+            case true => Iterator(true)
+            case false => Iterator.empty
+          }
+        }
```

Comment:
shall we create a method to define this df to save duplicated code?

Reply:
i have updated the PR with generic method to create df, is it ok now..?